### ISSUE TITLE
Error with kamakiri payload on mt6761 using macOS

### ISSUE THREAD

**Issue by michael-ford** [bkerler/mtkclient#198]
Trying to bootloader unlock on a Icon 2 with MT6761 A22 running android go edition 10 using macOS (11.2). Got the dependencies installed fine. Final exception when running `python mtk r` is `struct.error: unpack requires a buffer of 12 bytes`, but the first error is in the Kamakiri payload step. I suspect the problem is with `[Errno 13] Access denied (insufficient permissions)`, but I'm a little hesitant to run it using sudo right out of the gate so figured I'd ask here first.

Note I've installed all this in a conda environment, in case that is relevant. 

```Preloader - 	CPU:			MT6761/MT6762/MT3369/MT8766B(Helio A20/P22/A22/A25/G25)
Preloader - 	HW version:		0x0
Preloader - 	WDT:			0x10007000
Preloader - 	Uart:			0x11002000
Preloader - 	Brom payload addr:	0x100a00
Preloader - 	DA payload addr:	0x201000
Preloader - 	CQ_DMA addr:		0x10212000
Preloader - 	Var1:			0x25
Preloader - Disabling Watchdog...
Preloader - HW code:			0x717
Preloader - Target config:		0xe7
Preloader - 	SBC enabled:		True
Preloader - 	SLA enabled:		True
Preloader - 	DAA enabled:		True
Preloader - 	SWJTAG enabled:		True
Preloader - 	EPP_PARAM at 0x600 after EMMC_BOOT/SDMMC_BOOT:	False
Preloader - 	Root cert required:	False
Preloader - 	Mem read auth:		True
Preloader - 	Mem write auth:		True
Preloader - 	Cmd 0xC8 blocked:	True
Preloader - Get Target info
Preloader - BROM mode detected.
Preloader - 	HW subcode:		0x8a00
Preloader - 	HW Ver:			0xca01
Preloader - 	SW Ver:			0x200
Preloader - ME_ID:			FEDEBBD44E374AFFC5D1861C5BA18F20
Preloader - SOC_ID:			5901C30AF65F5563AE99C1FA8378D74D2D9935D58E4D598D3E9FDD562C89B903
PLTools - Loading payload from mt6761_payload.bin, 0x264 bytes
PLTools - Kamakiri / DA Run
Kamakiri - Trying kamakiri2..
[Errno 13] Access denied (insufficient permissions)
Kamakiri - Done sending payload...
PLTools - Error, payload answered instead:
Mtk
Mtk - [LIB]: Error on running kamakiri payload
DA_handler - Device is protected.
DA_handler - Device is in BROM mode. Trying to dump preloader.
Preloader
Preloader - [LIB]: Unknown: 0x1d0a
DAXFlash - Uploading xflash stage 1 from MTK_AllInOne_DA_5.2136.bin
Preloader
Preloader - [LIB]: DA_Send status error:SLA_ERROR (0x1d0d)
DAXFlash
DAXFlash - [LIB]: Error on sending DA.
DA_handler
DA_handler - [LIB]: Error uploading da
DA_handler - Requesting available partitions ....
Traceback (most recent call last):
  File "mtk", line 1381, in <module>
    mtk = Main(args).run()
  File "mtk", line 656, in run
    da_handler.handle_da_cmds(mtk, cmd, args)
  File "/Users/michaelford/Documents/mtkclient/mtkclient/Library/mtk_da_cmd.py", line 529, in handle_da_cmds
    self.da_read(partitionname=partitionname, parttype=parttype, filename=filename)
  File "/Users/michaelford/Documents/mtkclient/mtkclient/Library/mtk_da_cmd.py", line 138, in da_read
    gpttable = self.mtk.daloader.get_partition_data(parttype=parttype)
  File "/Users/michaelford/Documents/mtkclient/mtkclient/Library/mtk_daloader.py", line 169, in get_partition_data
    data, guid_gpt = self.da.partition.get_gpt(self.mtk.config.gpt_settings, parttype)
  File "/Users/michaelford/Documents/mtkclient/mtkclient/Library/partition.py", line 18, in get_gpt
    data = self.readflash(addr=0, length=2 * self.config.pagesize, filename="", parttype=parttype, display=False)
  File "/Users/michaelford/Documents/mtkclient/mtkclient/Library/mtk_daxflash.py", line 805, in readflash
    plen = self.get_packet_length()
  File "/Users/michaelford/Documents/mtkclient/mtkclient/Library/mtk_daxflash.py", line 726, in get_packet_length
    resp = self.send_devctrl(self.Cmd.GET_PACKET_LENGTH)
  File "/Users/michaelford/Documents/mtkclient/mtkclient/Library/mtk_daxflash.py", line 262, in send_devctrl
    status[0] = self.status()
  File "/Users/michaelford/Documents/mtkclient/mtkclient/Library/mtk_daxflash.py", line 213, in status```

**Comment by keerttttt** [comment]
You can try to use kamakiri instead of the default kamakiri2. I made a patch to make kamakiri 1 default on Mac, but I think it's not live yet.

**Comment by SmallPP420** [comment]
i actually got this issue try a different usb port

**Comment by keerttttt** [comment]
Should be fixed now, check #199

**Comment by michael-ford** [comment]
You all are legends. Apologies for not getting back sooner. Successfully bootloader unlocked and rooted using cli (tried gui hung at some stage) on the U325AC on mac using commit 0ad4402.
